src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -236,4 +237,42 @@ private function getRelationDataFromFormData($data)
 
         return $relationData;
     }
+
+    /**
+     * Transform the data before saving into the database. This is kind of an alias to Laravel Mutators but defined on Controler when defining the field.
+     * We basically call this "mutator" before saving the data into the database but after data is already validated.
+     *
+     * @param  array  $data  The form data.
+     * @return array The processed data.
+     */
+    protected function handleBeforeSavingCallback($data)
+    {
+        $fields = array_filter($this->fields(), function ($field) use ($data) {
+            return in_array($field['name'], array_keys($data)) && isset($field['beforeSaving']);
+        });
+
+        if (empty($fields)) {
+            return $data;
+        }
+        // cicle all the fields that have the beforeSaving callback
+        foreach ($fields as $field_name => $field_definition) {
+            if (! is_array($field_definition['name'])) {
+                $data[$field_name] = $this->runBeforeSavingCallback($data[$field_name], $field_name, $field_definition);
+            } else {
+                foreach ($field_definition['name'] as $f_name) {
+                    $data[$f_name] = $this->runBeforeSavingCallback($data[$f_name], $f_name, $field_definition);
+                }
+            }
+        }
+
+        return $data;
+    }
+
+    protected function runBeforeSavingCallback($current, $field_name, $field)
```

Comment:
I'm wondering - do we really need this function? Can't we just run `call_user_func()` above? I see no reason why we should get the current entry here, when they can get it in the callback itself if they need it. And if that goes away, this method can go away entirely, right?

Reply:
Not at all.

Just refactored it!